app/services/broadcasts/welcome_notification/generator.rb

```diff
@@ -11,19 +11,21 @@ def self.call(*args)
       end
 
       def call
-        return if send_welcome_notification || commented_on_welcome_thread?
+        return if commented_on_welcome_thread? || send_welcome_notification?
```

Comment:
Hm, `send_welcome_notification?` doesn't seemed correctly named here. Seems like we're saying "don't call this method if they commented on the welcome thread or if we have to send the welcome notification? 🤔

Reply:
That's a good point...how about something like `welcome_notification_exsits?` or `received_welcome_notification?` 🤔

Reply:
`has_received_notification?` seems clear to me. i think we're going to want to do this check for every welcome notification, so it would be nice for this method to be more generic. i can see this method taking in a user/user id and a broadcast/broadcast type, and then all this method really needs to be responsible for is checking whether that notification exists or not for the passed-in user and the passed-in broadcast.

does that make sense?

Reply:
Okay, I will change it to `has_received_welcome_notification?`, although I don't believe that rubocop likes `has` at the beginning of the method name--I had originally named `commented_on_welcome_thread?` as `has_commented_on_welcome_thread?` and rubocop did not like it. 😕 

I agree that we are going to want this to be more generic for future welcome notifications!

Okay, I'll keep that in mind as I make my changes. 🙂